### ISSUE TITLE
Melon is not working

### ISSUE THREAD

**Issue by octomelon** [Slyyxp/rsack#35]
Hi, I have subscription which is only available for streaming. But I can't download song using your [melon_client.py](https://gist.github.com/Slyyxp/b2a07a2756a5416bac4e9d03f85723ae#file-melon_client-py)
What I found is

- `auth()` through kakao is not working. link kakao account to melon id. And login through [https://member.melon.com/muid/mobile/login/login_login.json](url)
- https://spmd.melon.com/prestream only support http GET method, not POST
- Whenever I call funtion `streaming_path()`, It return mcache://smmd.melon.com/smartstream, not https://spmd.melon.com/prestream
- When I call https://play.melon.com/cds/delivery/android/streaming_path.json without q parameter which is obtained from streaming_path.json, It doesn't download song.
- After add q from streaming_path.json, I only download song that is encrypted

Here is [code!](https://gist.github.com/octomelon/d0e5b58b3cccb8eb296c4ecc036df735)

**Comment by Slyyxp** [comment]
This is not supposed to be functional code but rather a starting point for those interested in researching ripping from Melon.
To answer each of your points though

- I do recall `auth()` working correcly with Kakao as it's how I authenticated when previously testing. It has been a while though.
- POST'ing to https://spmd.melon.com/prestream with the provided data definitely responded with some form of preview track at the time of creation.
- Whether `cds/delivery/android/streaming_path.json` has been adjusted to respond with `/smartstream` or it's the data you provided I have no idea.
- There is no decryption module provided, nor do I know the encryption method.

To be more clear though, I have not looked at Melon in close to a year. I provided what little information I had for anyone else that may want to look at it further.
Although there is an issue out https://github.com/Slyyxp/rsack/issues/21 I don't really intend to follow through with it as of now as I am geoblocked from subscribing.

If you want to proceed with this you'll need to identify the encryption method on your own.
